ui/src/app/views/workspace-edit/component.tsx

```diff
@@ -0,0 +1,584 @@
+        }
+      }
+    }
+
+    resetWorkspaceEditor() {
+      this.setState({
+        workspaceCreationError : false,
+        workspaceCreationConflictError : false
+      });
+    }
+
+    isEmpty(field) {
+      const fieldValue = this.state.workspace[field];
+      return !fieldValue || fieldValue === '';
+    }
+
+    isMode(mode) {
+      return this.props.routeConfigData.mode === mode;
+    }
+
+    render() {
+      return <React.Fragment>
+        <div style={{width: '60%'}}>
+        <WorkspaceEditSection header={this.renderHeader()} tooltip={toolTipText.header}
+                              section={{marginTop: '24px'}} largeHeader required>
+          <div style={{display: 'flex', flexDirection: 'row'}}>
+            <TextInput type='text' style={styles.textInput} autoFocus placeholder='Workspace Name'
+                       value = {this.state.workspace.name}
+                       onChange={v => this.setState(fp.set(['workspace', 'name'], v))}/>
+            <TooltipTrigger
+                content='To use a different dataset version, clone or create a new workspace.'
+                disabled={!(this.isMode(WorkspaceEditMode.Edit))}>
+              <div style={styles.select}>
+                <Select style={{borderColor: 'rgb(151, 151, 151)', borderRadius: '6px'}}
+                        value={this.state.workspace.cdrVersionId}
+                        options={this.state.cdrVersionItems}
+                        onChange={v =>
+                            this.setState(fp.set(['workspace', 'cdrVersionId'], v))}
+                        isDisabled={this.isMode(WorkspaceEditMode.Edit)}>
+                </Select>
+              </div>
+            </TooltipTrigger>
+            <TooltipTrigger content={toolTipText.cdrSelect}>
+              <InfoIcon style={{...styles.infoIcon, marginTop: '0.5rem'}}/>
+            </TooltipTrigger>
+          </div>
+        </WorkspaceEditSection>
+        {this.isMode(WorkspaceEditMode.Clone) &&
+        <div style={{display: 'flex', flexDirection: 'row'}}>
+          <input type='checkbox'
+                 style={{height: '.66667rem', marginRight: '.31667rem', marginTop: '1.2rem'}}
+          onChange={v => this.setState({cloneUserRole: v.target.checked})}>
+              </input>
+          <WorkspaceEditSection header='Copy Original workspace Collaborators'
+            text='Share cloned workspace with same collaborators'/>
+        </div>
+        }
+        <WorkspaceEditSection header='Billing Account' subHeader='National Institutes of Health'
+            tooltip={toolTipText.billingAccount}
+            text='To fulfill program requirements, All of Us requests the following information
+            for each workspace.'/>
+        <WorkspaceEditSection header='Describe your research purpose'
+            tooltip={toolTipText.researchPurpose}
+            text={['Please include the ', <strong key='question'>research question</strong>,
```

Comment:
FYI: since the `text` prop just gets rendered, it can be anything renderable, including a `div` or `Fragment` which would probably be more convenient here.